src/sentry/ownership/grammar.py

```diff
@@ -201,7 +208,7 @@ def test_frames(
         ),
         match_frame_func: Callable[[Mapping[str, Any]], bool] = lambda _: True,
     ) -> bool:
-        for frame in (f for f in frames if isinstance(f, Mapping)):
```

Comment:
Removing the isinstance check here provides a _highly unscientific_ ~2-5% improvement on extremely large ownership files

Reply:
why are we able to remove that check?

Reply:
The frames parameter is defined as `Sequence[Mapping[str, Any]]`.

If we do the check in the loop, ex:
```python
for frame in frames:
    if not isinstance(frame, Mapping):
        continue
```
mypy will catch it as:
```
src/sentry/ownership/grammar.py:189: error: Statement is unreachable  [unreachable]
```

Not sure why mypy can't catch the inline version too but presumably it's just a missing optimization in mypy

Reply:
ah i see. okay. what a weird check.